eva-server/src/test/java/uk/ac/ebi/variation/eva/lib/datastore/DBAdaptorConnectorTest.java

```diff
@@ -0,0 +1,68 @@
+package uk.ac.ebi.variation.eva.lib.datastore;
+
+import com.mongodb.DB;
+import com.mongodb.MongoClient;
+import com.mongodb.ReadPreference;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import uk.ac.ebi.variation.eva.server.ApplicationConfiguration;
+
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Jose Miguel Mut Lopez &lt;jmmut@ebi.ac.uk&gt;
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = { ApplicationConfiguration.class })
+public class DBAdaptorConnectorTest {
+
+    @Autowired
+    private MongoDbFactory factory;
+
+    /**
+     * Check that spring is autowiring our MultiMongoDbFactory as the MongoDbFactory to use.
+     *
+     * To check it, we use MultiMongoDbFactory::setDatabaseNameForCurrentThread to change the DB we should get later
+     * when we do a `factory.getDb()`
+     */
+    @Test
+    public void testMongoDbFactoryAutowiring() {
+        String dbName = "DBAdaptorConnectorTest_testMongoDbFactoryAutowiring";
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(dbName);
+        DB db = factory.getDb();
+        assertEquals(db.getName(), dbName);
+    }
+
+    /**
+     * Check that the value:
+     * eva.mongo.read-preference=secondaryPreferred
+     * is properly read by DBAdaptorConnector::getMongoClient
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testDefaultReadPreferenceInMongoClient() throws Exception {
```

Comment:
This is not really testing the default now, since there shouldn't be any value set in the properties file in order to do that. In addition, it would be good to have a method that checks wrong paths, using manually created properties with some other fields missing.